Kernel/Heap/kmalloc.cpp

```diff
@@ -204,6 +204,14 @@ void kmalloc_enable_expand()
     g_kmalloc_global->allocate_backup_memory();
 }
 
+static inline void kmalloc_verify_nospinlock_held()
+{
+    // Catch bad callers allocating under spinlock.
+    if constexpr (KMALLOC_VERIFY_NO_SPINLOCK_HELD) {
+        VERIFY(!Processor::current().in_critical());
```

Comment:
Too late to the game, but can we change this to a call something like `SpinLock::any_may_be_held` instead? Reason for this is that SpinLocks actually shouldn't use critical sections at all, and instead just clear the interrupt flag (as is done in my SMP PR).